includes/class-wc-payments-order-service.php

```diff
@@ -2127,4 +2127,75 @@ private function is_order_type_object( $order ): bool {
 	private function intent_has_card_payment_type( $intent_data ): bool {
 		return isset( $intent_data['payment_method_type'] ) && 'card' === $intent_data['payment_method_type'];
 	}
+
+	/**
+	 * Countries where FROD balance is not supported.
+	 *
+	 * @var array
+	 */
+	const FROD_UNSUPPORTED_COUNTRIES = [ 'HK', 'SG', 'AE' ];
+
+	/**
+	 * Handle insufficient balance for refund.
+	 *
+	 * @param WC_Order $order  The order being refunded.
+	 * @param int      $amount The refund amount.
+	 */
+	public function handle_insufficient_balance_for_refund( WC_Order $order, $amount ) {
+		$account_country = WC_Payments::get_account_service()->get_account_country();
+
+		$formatted_amount = wc_price(
+			WC_Payments_Utils::interpret_stripe_amount( $amount, $order->get_currency() ),
+			[ 'currency' => $order->get_currency() ]
+		);
+
+		if ( $this->is_frod_supported( $account_country ) ) {
+			$order->add_order_note( $this->get_frod_support_note( $formatted_amount ) );
+		} else {
+			$order->add_order_note( $this->get_insufficient_balance_note( $formatted_amount ) );
+		}
+	}
+
+	/**
+	 * Check if FROD is supported for the given country.
+	 *
+	 * @param string $country_code Two-letter country code.
+	 * @return bool
+	 */
+	private function is_frod_supported( $country_code ) {
+		return ! in_array(
+			$country_code,
+			self::FROD_UNSUPPORTED_COUNTRIES,
+			true
+		);
+	}
+
+	/**
+	 * Get the order note for FROD supported countries.
+	 *
+	 * @param string $formatted_amount The formatted refund amount.
+	 * @return string
+	 */
+	private function get_frod_support_note( $formatted_amount ) {
+		return sprintf(
+			/* translators: %1$s: Formatted refund amount, %2$s: Link to FROD documentation */
+			__( 'Refund of %1$s failed due to insufficient funds in your WooPayments balance. To prevent delays in refunding customers, please consider adding funds to your Future Refunds or Disputes (FROD) balance. Learn more: %2$s', 'woocommerce-payments' ),
+			$formatted_amount,
+			'https://woocommerce.com/document/woopayments/fees-and-debits/preventing-negative-balances/#adding-funds'
```

Comment:
It would be nice if we could make this a link so merchants only need to click to get to the documentation. I think `WC_Payments_Utils::esc_interpolated_html` should help

Reply:
Good catch, addressed in https://github.com/Automattic/woocommerce-payments/pull/10313/commits/2eef38cb6c581099d54b5334aa75d5b91d48e70c